vendor-extra/JatsContentBundle/tests/EventListener/BuildView/RefListSectionListenerTest.php

```diff
@@ -0,0 +1,212 @@
+     */
+    public function it_sets_the_heading_and_content_arguments(string $xml, array $expectedArguments) : void
+    {
+        $translator = new Translator('en');
+        $translator->addLoader('array', new ArrayLoader());
+        $translator->addResource(
+            'array',
+            ['libero.page.references' => 'es string'],
+            'es',
+            'messages'
+        );
+
+        $listener = new RefListSectionListener($this->createDumpingConverter(), $translator);
+
+        $element = $this->loadElement($xml);
+
+        $event = new BuildViewEvent(
+            $element,
+            new TemplateView('@LiberoPatterns/section.html.twig', [], ['lang' => 'es'])
+        );
+        $listener->onBuildView($event);
+        $view = $event->getView();
+
+        $this->assertInstanceOf(TemplateView::class, $view);
+        $this->assertSame('@LiberoPatterns/section.html.twig', $view->getTemplate());
+        $this->assertEquals($expectedArguments, $view->getArguments());
+        $this->assertSame(['lang' => 'es', 'level' => 1], $view->getContext());
+    }
+
+    public function contentProvider() : iterable
+    {
+        yield 'no heading' => [
+            <<<XML
+<jats:ref-list xmlns:jats="http://jats.nlm.nih.gov"/>
+XML
+            ,
+            [
+                'heading' => [
+                    'text' => 'es string',
+                    'level' => 1,
+                ],
+                'content' => new TemplateView(
+                    '',
+                    [
+                        'node' => '/jats:ref-list',
+                        'template' => '@LiberoPatterns/reference-list.html.twig',
+                        'context' => ['lang' => 'es', 'level' => 2],
+                    ]
+                ),
+            ],
+        ];
+
+        yield 'heading' => [
+            <<<XML
+<jats:ref-list xmlns:jats="http://jats.nlm.nih.gov">
+    <jats:title>foo</jats:title>
+</jats:ref-list>
+XML
+            ,
+            [
+                'heading' => [
+                    'node' => '/jats:ref-list/jats:title',
+                    'template' => '@LiberoPatterns/heading.html.twig',
+                    'context' => ['lang' => 'es', 'level' => 1],
+                ],
+                'content' => new TemplateView(
+                    '',
+                    [
+                        'node' => '/jats:ref-list',
+                        'template' => '@LiberoPatterns/reference-list.html.twig',
+                        'context' => ['lang' => 'es', 'level' => 2],
+                    ]
+                ),
+            ],
+        ];
+    }
+
+    /**
+     * @test
+     */
+    public function it_uses_the_current_level() : void
+    {
+        $listener = new RefListSectionListener($this->createDumpingConverter(), new IdentityTranslator());
+
+        $element = $this->loadElement(
+            <<<XML
+<jats:ref-list xmlns:jats="http://jats.nlm.nih.gov">
+    <jats:title>Title</jats:title>
```

Comment:
is this a realistic example? I thought there would be `<ref>` elements inside `<ref-list>`

Reply:
looking at the expected values, know that elements inside here won't be used by this Listener, but would make the example complete (in documentation terms rather than coverage terms)